File: healthcare_app_server/chat_service/models/mongodb/group-member.js
const group_member_role = {
    "MEMBER": 0,
    "CO_OWNER": 1,
    "OWNER": 2
}

class GroupMember {
    constructor(id, groupId, userId) {
        this._role = "";
        this._created_at = this._updated_at = new Date();
        this._is_removed = false;
        this._role = group_member_role.MEMBER;
        this._id = id;
        this._groupId = groupId;
        this._userId = userId;
    }

    get id() {
        return this._id;
    }

    get groupId() {
        return this._groupId;
    }

    get userId() {
        return this._userId;
    }

    get created_at() {
        return this._created_at;
    }

    get updated_at() {
        return this._updated_at;
    }

    get is_removed() {
        return this._is_removed;
    }

    get role() {
        return this._role;
    }
}

File: healthcare_app_server/admin_service/src/main/java/dev/skyherobrine/service/models/mongodb/GroupMember.java
package dev.skyherobrine.service.models.mongodb;

import com.fasterxml.jackson.annotation.JsonFormat;
import dev.skyherobrine.service.enums.GroupMemberRole;
import lombok.*;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;
import org.springframework.data.mongodb.core.mapping.FieldType;
import org.springframework.data.mongodb.core.mapping.MongoId;

import java.time.LocalDateTime;

@Document(collection = "group_members")
@Getter @Setter @NoArgsConstructor @RequiredArgsConstructor
public class GroupMember {
    @MongoId @Field(targetType = FieldType.INT64) @NonNull
    private Long id;
    @Field(targetType = FieldType.STRING) @NonNull
    private String groupId;
    @Field(targetType = FieldType.STRING) @NonNull
    private String userId;
    @Field(targetType = FieldType.STRING) @NonNull
    private GroupMemberRole role;
    @JsonFormat(pattern = "dd-MM-yyyy-HH-mm-ss")
    @Field(targetType = FieldType.DATE_TIME) @NonNull
    private LocalDateTime createdAt;
    @JsonFormat(pattern = "dd-MM-yyyy-HH-mm-ss")
    @Field(targetType = FieldType.DATE_TIME) @NonNull
    private LocalDateTime updatedAt;
    @Field(targetType = FieldType.BOOLEAN) @NonNull
    private Boolean isRemoved;
}
